### PR TITLE
Fix return value is the function signature

### PR DESCRIPTION
**Pull request by steffenix**

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1,2 +1 @@
-erlang 25.0.2
```
Comment (alisinabh) [inline .tool-versions:1]: I believe this should not be in this PR.

File: lib/abi/type_decoder.ex [hunk 2]
```diff
@@ -151,9 +151,16 @@ defmodule ABI.TypeDecoder do
...
       when is_binary(method_id) do
-    case ABI.Util.split_method_id(encoded_data) do
-      {:ok, ^method_id, rest} -> decode_raw(rest, types)
-      _ -> decode_raw(encoded_data, types)
+    if rem(byte_size(encoded_data), 32) == 0 do
+      decode_raw(encoded_data, types)
+    else
+      case ABI.Util.split_method_id(encoded_data) do
+        {:ok, ^method_id, rest} ->
+          decode_raw(rest, types)
```
Comment (alisinabh) [inline lib/abi/type_decoder.ex:159]: I actually have never seen the output of a call to include the method id. Is this even needed?

Comment (pandadefi) [reply lib/abi/type_decoder.ex:159]: Look at the associated issue, in the case of erc-1271, the `isValidSignature`function returns 0x1626ba7e which is the function signature.

https://github.com/poanetwork/ex_abi/issues/173

File: .tool-versions (hunk 1 above)
Comment (steffenix) [reply .tool-versions]: Updated.

File: lib/abi/type_decoder.ex (hunk 2 above)
Comment (steffenix) [reply lib/abi/type_decoder.ex:159]: I think the split method id can be removed yes.

**Review by ayrat555 — COMMENTED** [review]
will it be possible to add at least one test?

**Review by ayrat555 — APPROVED** [review]

**Pull request merged**